c10/core/TensorImpl.cpp

```diff
@@ -29,16 +29,18 @@ const char * const TensorImpl::err_msg_tensor_metadata_change_not_allowed =
     "        x.set_(y)";
 
 at::Tensor& TensorImpl::grad() {
-  if (autograd_meta()) {
-    return autograd_meta()->grad();
+  if (type_set_.has(TensorTypeId::VariableTensorId)) {
```

Comment:
This could be wrapped in a nice function to make code more readable?
I'm fine if you don't do it because it will go away with the next commit.

Reply:
Yes, it goes away next commit :>